src/Rector/Property/ProphecyPHPDocRector.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PHPUnit\Rector\Property;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\Stmt\Use_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\PHPUnit\NodeAnalyzer\TestsNodeAnalyzer;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\PHPUnit\Tests\Rector\Property\ProphecyPHPDocRector\ProphecyPHPDocRectorTest
+ */
+class ProphecyPHPDocRector extends AbstractRector
+{
+    public function __construct(
+        private readonly TestsNodeAnalyzer $testsNodeAnalyzer
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Add correct @var to ObjectProphecy instances based on $this->prophesize() call.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class HelloTest extends TestCase
+{
+    /**
+     * @var SomeClass
+     */
+    private $test;
+
+    public function setUp(): void
+    {
+        $this->test = $this->prophesize(SomeClass::class);
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class HelloTest extends TestCase
+{
+    /**
+     * @var ObjectProphecy<SomeClass>
+     */
+    private $test;
+
+    public function setUp(): void
+    {
+        $this->test = $this->prophesize(SomeClass::class);
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        // PhpParser\Node\Expr\Assign
+        // PhpParser\Node\Expr\MethodCall
+        return [Assign::class];
```

Comment:
The best practise while working with AST is to hook into the node that is being changed. 
Otherwise you might try to change nodes, that were already traversed and changed.

Here you change the `Property` doc/type, so you should hook into `Class_` node :+1:

Reply:
I will not be able todo this in the next time, if somebody want to take it over and convert it to that I'm fine.